File: 浦发机考真题/1.判断素数.py
# 题目描述
# * 判断素数：除了1和该数本身，没有其他可以整除的数（素数最小为2）
import math
def is_prime(n):
            if n<2: return False
            if n==2: return True
            for i in range(2, int(math.sqrt(n))+1):
                if n%i==0:
                    return False
            return True

# 简写
def isPrime(n):
    return n>1 and all(n%i!=0 for i in range(2,int(n**.5)+1))


# TDD - 测试
"""
100以内的质数(素数)有2，3，5，7，11，13，17，19，23，29，31，37，41，
43，47，53，59，61，67，71，73，79，83，89，97，在100内共有25个质数。
"""
for i in range(1,101):
    # if is_prime(i):
    if isPrime(i):
        print(i)

File: 浦发机考编程题/1.判断素数.java
// 输入一个正整数，判断是否为素数
// 素数：除了1和它本身以外，没有被其他的数可以被整除（最小的素数为2）

import java.util.Scanner;

public class Main{
    
    // isPrime()
    public static boolean isPrime(int n){
        // 为什么只需要穷举到sqrt(n)+1?
        // 因为：可以数学证明 https://www.zhihu.com/question/21808179
        if (n < 2){return false;} // 最小的素数为2
        if (n==2){return true;}
        for (int i=2; i<Math.sqrt(n)+1; i++){
            if (n % i == 0){
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        System.out.println("input a interger: ");
        int n = sc.nextInt();
        // 只需要：从1穷举到sqrt(n)+1（可以数学上严格证明）
        System.out.println("result: " + isPrime(n));
    }
}